datamodel.customer-survey.xml

```diff
@@ -1922,6 +1925,64 @@ EOF
 				throw new Exception(Dict::S('Survey-email-notsent').': '.implode(', ', $aErrors));
 		}
 	}]]></code>
+        </method>
+        <method id="ApplyParamsToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param Contact $oContact
+	 * @param ?string $sQuizzUrl
+	 *
+	 * @return string Content with placeholders replaced.
+	 */</comment>
+          <static>false</static>
+          <access>protected</access>
+          <type>Custom</type>
+          <code><![CDATA[	protected function ApplyParamsToContent($sContent, $oContact, $sQuizzUrl)
+{
+  $aParams = [];
+  $aParams['recipient->object()'] = $oContact;
+  if (false === is_null($sQuizzUrl)){
+    $aParams['quiz_url'] = $sQuizzUrl;
+  }
+  $aParams['this->object()'] = $this;
+
+  $sContent = MetaModel::ApplyParams($sContent, $aParams);
+
+  $this->AppendLinkToContent($sContent, $sQuizzUrl);
+
+  return $sContent;
+}
+]]></code>
+        </method>
+        <method id="AppendLinkToContent">
+          <comment>/**
+	 * @param string $sContent
+	 * @param ?string $sQuizzUrl
+	 *
+	 * @return string If content contains a quiz_url placeholder, and url param set then Content with link appended, otherwise unchanged content
+*/</comment>
+          <static>true</static>
+          <access>public</access>
+          <type>Custom</type>
+          <code><![CDATA[	public static function AppendLinkToContent($sContent, $sQuizzUrl)
+{
+    if (is_null($sQuizzUrl)) {
+        return $sContent;
+    }
+
+    $quizzUrlPatternsFound = preg_match('/\$quiz_url\$/', $sContent);
```

Comment:
I actually shorted this based on @Hipska 's suggestion; why should we make it longer again?

Reply:
I made those modifications for 2 reasons : 
- I prefer having an explicit variable name than a variable name + an inline comment
- preg_match doesn't return a boolean (1, 0 or false), so I prefer using a true boolean in the test

Reply:
I agree the variable name, the rest can be simplified as:

```php
     if ($quizUrlPatternsFound === 1) {
         return $sContent;
     }

     return $sContent . '<br/><a href="'.$sQuizUrl.'">'.Dict::S('Survey-notif-linktoquizz').'</a>';
```

Reply:
Thanks, really like the idea ! 
Pushed in 1328f76d